pkg/admission/types.go

```diff
@@ -31,3 +32,9 @@ type WantsInformerFactory interface {
 	SetInformerFactory(informers.SharedInformerFactory)
 	Validator
 }
+
+// WantsAuthorizer defines a function which sets Authorizer for admission plugins that need it.
+type WantsAuthorizer interface {
+	SetAuthorizer(authorizer.Authorizer)
+	Validator
```

Comment:
> So you're saying you're going to have complex initialization interactions and validation spread across N setters and a separate validator? Why not at this point just create a config struct that can be easily added to and have all the interaction and validation in one method/spot?

Having an initializer interfaces allows someone who chooses to extend the admission chain via godep-ing a spot to add custom initializers without having to patch upstream code (a config struct as a for instance).